### PR TITLE
feat(cli): update clap to v4

### LINKED ISSUE #157: Clap v4
Update from clap v3 to clap v4, which provides an easier API and changes the output of help, among others. See https://epage.github.io/blog/2022/09/clap4/

### PR DESCRIPTION
**Pull request by G3root**
## About

fixes #157

### CODE REVIEW

File: .changeset/silent-shrimps-dance.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@lagon/cli': minor
+---
+
+update clap to v4
```
Comment (QuiiBz) [inline .changeset/silent-shrimps-dance.md:5]: Could you capitalize `update`?

File: packages/cli/src/main.rs [hunk 2]
```diff
@@ -9,7 +9,7 @@ mod commands;
 mod utils;
 
 #[derive(Parser, Debug)]
-#[clap(author, version, about, long_about = None, arg_required_else_help = true)]
+#[command(author, version, about, long_about = None, arg_required_else_help = true)]
```
Comment (QuiiBz) [inline packages/cli/src/main.rs:12]: Happy to see only one change, I expected more than that.

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks for the PR! Just a wording change and we'll ready to merge/

**Review by QuiiBz — APPROVED** [review]
LGTM, thanks a lot!

**Comment by G3root** [comment]
> LGTM, thanks a lot!

open some good first issues, i'll help with some.

**Pull request merged**